### PR TITLE
Add CholeskySolveVec method and auxiliary routines

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by kortschak** [comment]
LGTM

File: mat64/cholesky.go [hunk 1]
```diff
@@ -79,8 +79,7 @@ func (t *TriDense) Cholesky(a *SymDense, upper bool) (ok bool) {
 }
 
 // SolveCholesky finds the matrix x that solves A * X = B where A = L * L^T or
-// A = U^T * U, and U or L are represented by t. The matrix A must be symmetric
-// and positive definite.
+// A = U^T * U, and U or L are represented by t.
 func (m *Dense) SolveCholesky(t Triangular, b Matrix) {
```
Comment (vladimir-ch) [inline mat64/cholesky.go]: Dense receivers are almost consistently called 'm' but for example in this case wouldn't it be better to relate better the function signature with the comment? The comment talks about x, X, A, B, L, U, while the function takes m, t, b.

File: mat64/cholesky.go [hunk 2]
```diff
@@ -109,6 +108,31 @@ func (m *Dense) SolveCholesky(t Triangular, b Matrix) {
 	}
 }
 
+// SolveCholesky finds the vector x that solves A * x = b where A = L * L^T or
```
Comment (vladimir-ch) [inline mat64/cholesky.go]: SolveCholeskyVec

File: mat64/vector.go [hunk 3]
```diff
@@ -103,6 +105,15 @@ func (m *Vector) RawVector() blas64.Vector {
 	return m.mat
 }
 
+// Copy makes a copy of elements of a into the receiver. It is similar to the
```
Comment (vladimir-ch) [inline mat64/vector.go]: CopyVec

File: mat64/cholesky.go (hunk 1 above)
Comment (kortschak) [inline mat64/cholesky.go]: Add ", placing the result in the receiver."

Comment (vladimir-ch) [inline mat64/cholesky.go]: ... and make both x in the comment the same case.

File: mat64/cholesky.go [hunk 4]
```diff
@@ -109,6 +108,31 @@ func (m *Dense) SolveCholesky(t Triangular, b Matrix) {
...
+// A = U^T * U, and U or L are represented by t.
```
Comment (vladimir-ch) [inline mat64/cholesky.go]: ", placing the result in the receiver." also here?

Comment (kortschak) [inline mat64/cholesky.go]: Yes, I think so.

File: mat64/vector.go (hunk 3 above)
Comment (btracey) [inline mat64/vector.go]: Done

File: mat64/cholesky.go (hunk 4 above)
Comment (btracey) [inline mat64/cholesky.go]: Done.

File: mat64/cholesky.go (hunk 2 above)
Comment (btracey) [inline mat64/cholesky.go]: Done.

File: mat64/cholesky.go (hunk 1 above)
Comment (btracey) [inline mat64/cholesky.go]: Done.

**Pull request merged**